modules/external-deployments/operator_role_policies.tf

```diff
@@ -0,0 +1,205 @@
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogStreams",
+      "logs:GetLogEvents",
+      "logs:FilterLogEvents"
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:logs:${local.region}:${local.account_id}:log-group:/aws/sagemaker/*"
+    ]
+  }
+  statement {
+    sid    = "SagemakerManageResources"
+    effect = "Allow"
+    actions = [
+      "sagemaker:AddTags",
+      "sagemaker:CreateEndpoint",
+      "sagemaker:CreateEndpointConfig",
+      "sagemaker:CreateModel",
+      "sagemaker:DeleteEndpoint",
+      "sagemaker:DeleteEndpointConfig",
+      "sagemaker:DeleteModel",
+      "sagemaker:DescribeEndpoint",
+      "sagemaker:DescribeEndpointConfig",
+      "sagemaker:DescribeModel",
+      "sagemaker:InvokeEndpoint",
+      "sagemaker:UpdateEndpointWeightsAndCapacities"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "AutoscalingForSagemaker"
+    effect = "Allow"
+    actions = [
+      "application-autoscaling:DeleteScalingPolicy",
+      "application-autoscaling:DeregisterScalableTarget",
+      "application-autoscaling:DescribeScalableTargets",
+      "application-autoscaling:DescribeScalingActivities",
+      "application-autoscaling:DescribeScalingPolicies",
+      "application-autoscaling:PutScalingPolicy",
+      "application-autoscaling:RegisterScalableTarget",
+      "application-autoscaling:TagResource"
+    ]
+    resources = ["*"]
+  }
+  statement {
+    sid    = "CloudwatchForAutoscaling"
+    effect = "Allow"
+    actions = [
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:DeleteAlarms",
+      "cloudwatch:DescribeAlarms"
+    ]
+    resources = [
+      "*"
+    ]
+  }
+  statement {
+    sid    = "IamAllowCreateServiceLinkedRole"
+    effect = "Allow"
+    actions = [
+      "iam:CreateServiceLinkedRole"
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:iam::${local.account_id}:role/aws-service-role/sagemaker.application-autoscaling.amazonaws.com/*"
+    ]
+    condition {
+      test     = "StringLike"
+      variable = "iam:AWSServiceName"
+      values = [
+        "sagemaker.application-autoscaling.amazonaws.com"
+      ]
+    }
+  }
+  statement {
+    sid    = "EcrRegistryReadDominoEnvironments"
+    effect = "Allow"
+    actions = [
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:BatchGetImage",
+      "ecr:GetDownloadUrlForLayer",
+    ]
+    resources = [
+      "arn:${data.aws_partition.current.partition}:ecr:${local.region}:${local.account_id}:repository/${local.environments_repository}",
+      "arn:${data.aws_partition.current.partition}:ecr:${local.region}:${local.account_id}:repository/${local.environments_repository}*"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "assume_any_role" {
+  statement {
+    sid       = "StsAllowOtherAccountsAssumeRole"
+    effect    = "Allow"
+    actions   = ["sts:AssumeRole"]
+    resources = ["*"]
+    condition {
```

Comment:
@steved I did some testing and this restriction *does* keep the role from assuming another role in the same account.
I tested with the `smtest61590-eks` role.  When I removed this condition the `smtest61590-external-deployments-operator` role was able to assume `smtest61590`.  When I have this condition in here the assume role fails with
```
An error occurred (AccessDenied) when calling the AssumeRole operation: User: arn:aws:sts::399181051250:assumed-role/smtest61590-external-deployments-operator/botocore-session-1726579049 is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::399181051250:role/smtest61590-eks
```

Reply:
So I think this should satisfy the concern about not being able to assume any other role in the same AWS account while still enabling this role to assume a role in another account without changes in this AWS account?